cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/RunnableTaskLauncher.java

```diff
@@ -31,21 +34,24 @@ public RunnableTaskLauncher(CConfiguration cConf) {
   }
 
   public byte[] launchRunnableTask(RunnableTaskRequest request) throws Exception {
-
-    ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-    if (classLoader == null) {
-      classLoader = getClass().getClassLoader();
-    }
+    ClassLoader classLoader = getClassLoader();
+    Injector injector = Guice.createInjector(new RunnableTaskModule(cConf));
     Class<?> clazz = classLoader.loadClass(request.getClassName());
-
-    Object obj = clazz.getDeclaredConstructor().newInstance();
-
+    Object obj = injector.getInstance(clazz);
```

Comment:
Don't create the instance and check the `instanceof`. Use `RunnableTask.class.isAssignableFrom(clazz)` to check before creating the isntance.

Reply:
Also, why use guice to create the task instance? I think it should always be created with the default constructor. Everything the task needed should be provided via the `RunnableTaskContext`.

Reply:
Done